tests/components/advantage_air/test_cover.py

```diff
@@ -0,0 +1,113 @@
+"""Test the Advantage Air Cover Platform."""
+
+from json import loads
+
+from homeassistant.components.advantage_air.const import (
+    ADVANTAGE_AIR_STATE_CLOSE,
+    ADVANTAGE_AIR_STATE_OPEN,
+)
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    DEVICE_CLASS_DAMPER,
+    DOMAIN as COVER_DOMAIN,
+    SERVICE_CLOSE_COVER,
+    SERVICE_OPEN_COVER,
+    SERVICE_SET_COVER_POSITION,
+)
+from homeassistant.const import ATTR_ENTITY_ID, STATE_OPEN
+
+from tests.components.advantage_air import (
+    TEST_SET_RESPONSE,
+    TEST_SET_URL,
+    TEST_SYSTEM_DATA,
+    TEST_SYSTEM_URL,
+    add_mock_config,
+)
+
+
+async def test_cover_async_setup_entry(hass, aioclient_mock):
+    """Test climate setup without sensors."""
+
+    aioclient_mock.get(
+        TEST_SYSTEM_URL,
+        text=TEST_SYSTEM_DATA,
+    )
+    aioclient_mock.get(
+        TEST_SET_URL,
+        text=TEST_SET_RESPONSE,
+    )
+
+    await add_mock_config(hass)
+
+    registry = await hass.helpers.entity_registry.async_get_registry()
+
+    assert len(aioclient_mock.mock_calls) == 1
+
+    # Test Cover Zone Entity
+    entity_id = "cover.zone_open_without_sensor"
+    state = hass.states.get(entity_id)
+    assert state
+    assert state.state == STATE_OPEN
+    assert state.attributes.get("device_class") == DEVICE_CLASS_DAMPER
+    assert state.attributes.get("current_position") == 100
+
+    entry = registry.async_get(entity_id)
+    assert entry
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)